src/main/java/fr/sncf/osrd/railjson/parser/RailJSONParser.java

```diff
@@ -68,9 +68,20 @@ public static Infra parse(RJSInfra railJSON) throws InvalidInfraException {
         var switchNames = new HashMap<String, Switch>();
         var switchIndex = 0;
         for (var rjsSwitch : railJSON.switches) {
-            var index = nodeIDs.get(rjsSwitch.base);
-            switchNames.put(rjsSwitch.id, trackGraph.makeSwitchNode(index, rjsSwitch.id, switchIndex++,
-                    rjsSwitch.positionChangeDelay));
+            // get the first port of the switch that will be consider as the base
+            var base = rjsSwitch.ports.entrySet().stream().findFirst().get().getValue();
```

Comment:
We should test that the map isn't empty and throw an InvalidInfraException otherwise. I don't think a NoSuchElementException would be properly forwarded to the middle/front end.